lighthouse-core/config/seo.js

```diff
@@ -12,10 +12,17 @@ module.exports = {
     gatherers: [
       'seo/meta-description',
     ],
+  }, {
+    passName: 'extraPass',
+    gatherers: [
+      'seo/font-size',
+      'styles',
```

Comment:
unfortunate that this relies on the styles gatherer :/ do you foresee a way the audit could optionally rely on the styles gatherer output? i.e. still report the violations but maybe with less helpful identifiers? I'm not sure we're going to get to a place in DevTools where we want to add the extra pass by default and it'd be a shame to miss out on the whole audit

Reply:
Only thing I need the styles gatherer for is `stylesheetId -> stylesheet URL` info. We can just say 'external stylesheet' if we don't have that mapping info, but I wonder - what's wrong with the 'styles' gatherer? Why does it need a separate run? Maybe I can collect `stylesheetId -> stylesheet URL` information in my gatherer separately and avoid requiring a separate run?

Reply:
styles gatherer parses all the stylesheets on the page with gonzales which can be *extremely* slow in some cases (old runs of theverge spent ~20s in styles gatherer)

this seems like a useful split though for cases where you don't need the actual parsed content of stylesheets 👍

Reply:
I've tried creating a `styles-metadata` gatherer that duplicates `styles` gatherer minus the parsing. However, I couldn't get stylSheetIds to match. `font-size` gatherer returned different IDs than `styles-metadata` gatherer for the exact same stylesheets. As far as I can see, this happens when you have two separate `CSS.enable->CSS.disable` runs. When I've put `CSS.styleSheetAdded` into the `font-size` gatherer (same `CSS.enable->CSS.disable` run) everything started to work.

Why this happens? No idea, possibly a Chrome bug. Why separate runs worked when I used `styles` gatherer? This gatherer starts collecting stylesheet info in `beforePass` and finishes in `afterPass`, I can't do it in the default run because LH will complain that: "CSS domain enabled when starting trace".

I'm not a fan of keeping `CSS.styleSheetAdded` in the `font-size` gatherer, but don't see any other option at this point. Please let me know if you have any other ideas.